daemon/execdriver/lxc/driver.go

```diff
@@ -324,24 +324,20 @@ func (d *Driver) Run(c *execdriver.Command, pipes *execdriver.Pipes, hooks execd
 
 	c.ContainerPid = pid
 
-	oomKill := false
-	oomKillNotification, err := notifyOnOOM(cgroupPaths)
-
 	if hooks.Start != nil {
 		logrus.Debugf("Invoking startCallback")
-		hooks.Start(&c.ProcessConfig, pid, oomKillNotification)
-
+		chOOM := make(chan struct{})
+		close(chOOM)
+		hooks.Start(&c.ProcessConfig, pid, chOOM)
 	}
 
+	oomKillNotification := notifyChannelOOM(cgroupPaths)
```

Comment:
This can't be moved before the start hook for this reason:
```
WARN[0007] lstat /sys/fs/cgroup/memory/lxc/252827e4c2f93d1b1e2127c385a2cd677fc76714fece570794f2470ae2065a79/cgroup.event_control: no such file or directory
```
I've added logrus call in the event fd goroutine below cause this error was hidden